src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -759,6 +760,34 @@ PurgeConnectionForPlacement(ShardPlacement *placement)
 
 
 /*
+ * Simply removes a given connection from the transaction participant hash.
+ */
+static void
+RemoveXactConnection(PGconn *connection)
```

Comment:
Yes, but I wasn't sure a refactor was in order, given it's exactly one duplication.